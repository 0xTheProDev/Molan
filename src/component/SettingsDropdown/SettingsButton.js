// @flow
import React from 'react';
import { Button, Icon } from 'semantic-ui-react';

export const SettingsButton = (
  <Button floated="right" title="Settings" basic>
    <Icon name="settings" />
  </Button>
);
