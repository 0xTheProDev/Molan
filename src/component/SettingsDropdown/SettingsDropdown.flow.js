// @flow
import { SettingsOption } from 'types/editor.flow';

export type SettingsDropdownProps = {
  SettingsButtonTitle: string,
  SettingsOptions: Array<SettingsOption>,
};
