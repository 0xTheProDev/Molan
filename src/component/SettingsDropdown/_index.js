// @flow
export { SettingsDropdown, SettingsDropdown as default } from './SettingsDropdown';
export { SettingsDropdownProps } from './SettingsDropdown.flow';
