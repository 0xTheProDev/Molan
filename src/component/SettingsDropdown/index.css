.ui.checkbox {
    display: block !important;
    padding: 5px 0 !important;
}