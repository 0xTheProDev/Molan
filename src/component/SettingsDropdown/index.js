import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Popup, Checkbox, Button, Icon, Header } from 'semantic-ui-react';
import './index.css';

export default class SettingsDropdown extends Component {
    static propTypes = {
        darkThemed:       PropTypes.bool.isRequired,
        onChangeLineNum:  PropTypes.func.isRequired,
        onChangeTheme:    PropTypes.func.isRequired,
        isFullScreen:     PropTypes.bool.isRequired,
        onFullScreen:     PropTypes.func.isRequired
    };

    render() {
        const { darkThemed, onChangeTheme, isFullScreen, onFullScreen } = this.props;

        return (
            <div>
                <Popup
                  trigger={<Button floated='right' title='Settings' basic><Icon name='settings'/></Button>}
                  on='click'
                  position='bottom center'
                >
                    <Header as='h5'>Settings</Header>
                    <Checkbox
                      label="Night Mode"
                      checked={darkThemed}
                      onChange={onChangeTheme}
                    />
                    <Checkbox
                      label="Show Line Numbers"
                      checked={this.props.lineNumbers}
                      onChange={this.props.onChangeLineNum}
                    />
                    <Checkbox
                      label="Show Minimap"
                      checked={this.props.minimap}
                      onChange={this.props.onChangeMinimap}
                    />
                </Popup>
                <Button basic
                  floated='right'
                  title='Full Screen'
                  active={isFullScreen}
                  onClick={onFullScreen}
                >
                    <Icon name='expand'/>
                </Button>
            </div>
        );
    }
}
