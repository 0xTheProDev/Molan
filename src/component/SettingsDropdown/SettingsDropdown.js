// @flow
import React from 'react';
import { Button, Checkbox, Popup, Header } from 'semantic-ui-react';

import { SettingsDropdownProps } from './SettingsDropdown.flow';

export function SettingsButton({ SettingsButtonTitle }: SettingsDropdownProps) {
  return (
    <Button floated="right" title={SettingsButtonTitle} basic>
      <Icon name="settings" />
    </Button>
  );
}

export function SettingsOption({ SettingsOptions }: SettingsDropdownProps) {
  return SettingsOptions.map(settingsOption => 
    <Checkbox
      key={settingsOption.key}
      label={settingsOption.label}
      checked={settingsOption.state}
      onChange={settingsOption.toggleStatus}
    />  
  );
}

export function SettingsDropdown(props: SettingsDropdownProps) {
  const { SettingsButtonTitle } = props;

  return (
    <Popup
      trigger={<SettingsButton {...props} />}
      on="click"
      position="bottom center"
    >
      <Header as="h5">{SettingsButtonTitle}</Header>
      {SettingsOption(props)}
    </Popup>
  );
}

SettingsDropdown.defaultProps = {
  SettingsButtonTitle: 'Settings',
  SettingsOptions: [],
};
