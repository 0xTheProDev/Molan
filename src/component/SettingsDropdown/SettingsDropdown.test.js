import React from 'react';

import { SettingsDropdown } from './SettingsDropdown';

describe('Settings Dropdown', () => {
  let
    onChangeThemeStub,
    onChangeLineNumStub,
    onChangeMinimapStub,
    onFullScreenStub,
    minProps;

  beforeAll(() => {
    onChangeThemeStub = jest.fn();
    onChangeLineNumStub = jest.fn();
    onChangeMinimapStub = jest.fn();
    onFullScreenStub = jest.fn();

    minProps = {
      darkThemed: false,
      onChangeTheme: onChangeThemeStub,
      lineNumbers: false,
      onChangeLineNum: onChangeLineNumStub,
      minimap: false,
      onChangeMinimap: onChangeMinimapStub,
      isFullScreen: false,
      onFullScreen: onFullScreenStub,
    };
  });

  it('should render', () => {
    const wrapper = shallow(<SettingsDropdown {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
