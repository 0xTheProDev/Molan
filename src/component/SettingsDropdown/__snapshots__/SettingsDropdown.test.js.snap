// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Dropdown should render 1`] = `
<div>
  <Popup
    disabled={false}
    eventsEnabled={true}
    offset={0}
    on="click"
    pinned={false}
    position="bottom center"
    trigger={
      <Button
        as="button"
        basic={true}
        floated="right"
        title="Settings"
      >
        <Icon
          as="i"
          name="settings"
        />
      </Button>
    }
  >
    <Header
      as="h5"
    >
      Settings
    </Header>
    <Checkbox
      checked={false}
      label="Night Mode"
      onChange={[MockFunction]}
      type="checkbox"
    />
    <Checkbox
      checked={false}
      label="Show Line Numbers"
      onChange={[MockFunction]}
      type="checkbox"
    />
    <Checkbox
      checked={false}
      label="Show Minimap"
      onChange={[MockFunction]}
      type="checkbox"
    />
  </Popup>
  <Button
    active={false}
    as="button"
    basic={true}
    floated="right"
    onClick={[MockFunction]}
    title="Full Screen"
  >
    <Icon
      as="i"
      name="expand"
    />
  </Button>
</div>
`;
